src/NuGet.Core/NuGet.Common/NuGetEnvironment.cs

```diff
@@ -44,15 +51,42 @@ public static string GetFolderPath(NuGetFolderPath folder)
             }
         }
 
-#if DNXCORE50
-        private static class Environment
+        private static string GetHome()
         {
-            public static string NewLine { get; } = System.Environment.NewLine;
+            if (RuntimeEnvironmentHelper.IsWindows)
+            {
+                return FirstNonEmpty(
+                    () => GetEnvironmentVariable("USERPROFILE"),
+                    () => GetEnvironmentVariable("HOMEDRIVE") + GetEnvironmentVariable("HOMEPATH"));
+            }
+            else
+            {
+                return GetEnvironmentVariable("HOME");
+            }
+        }
 
-            public static string GetEnvironmentVariable(string variable)
+        public static string GetEnvironmentVariable(string variable)
+        {
+            return System.Environment.GetEnvironmentVariable(variable);
+        }
+
+        private static string FirstNonEmpty(params Func<string>[] providers)
+        {
+            foreach (var p in providers)
             {
-                return System.Environment.GetEnvironmentVariable(variable);
+                var value = p();
+                if (!string.IsNullOrEmpty(value))
+                {
+                    return value;
+                }
             }
+            return null;
+        }
+
+#if DNXCORE50
+        private static class Environment
```

Comment:
Which is why in want of simplified. What we have is a hack, makes the code hard to follow and like you said yourself "not quite sure why it is there"